### PR TITLE
Reduce search radius for local minimization

### PR DESCRIPTION
**Pull request by proteneer**
Switch to L-BFGS-B and relax the assertion into a warning if we failed to minimize.

### CODE REVIEW

**Review by maxentile — APPROVED** [review]

File: timemachine/md/minimizer.py [hunk 1]
```diff
@@ -349,13 +350,17 @@ def val_and_grad_fn_bfgs(x_local_flattened):
...
+    )
 
     x_final = x0.copy()
     x_final[local_idxs] = res.x.reshape(x_local_shape)
 
     u_final, _ = val_and_grad_fn(x_final)
 
-    assert u_final < u_0, f"u_0: {u_0:.3f}, u_f: {u_final:.3f}"
+    if u_final >= u_0:
+        warnings.warn(f"Energy minimization with L-BFGS-B failed: u_0: {u_0:.3f}, u_f: {u_final:.3f}")
```
Comment (jkausrelay) [inline timemachine/md/minimizer.py:364]: would it make sense to return the original coords in this case?

Comment (proteneer) [reply timemachine/md/minimizer.py:364]: probably good just in case

**Review by jkausrelay — APPROVED** [review]

File: timemachine/md/minimizer.py [hunk 2]
```diff
@@ -349,13 +350,18 @@ def val_and_grad_fn_bfgs(x_local_flattened):
     x_local_0 = x0[local_idxs]
     x_local_0_flat = x_local_0.reshape(-1)
 
-    res = scipy.optimize.minimize(val_and_grad_fn_bfgs, x_local_0_flat, jac=True, options={"disp": True})
+    # use limited memory L-BFGS-B even though we don't have bounds
+    res = scipy.optimize.minimize(
+        val_and_grad_fn_bfgs, x_local_0_flat, method="L-BFGS-B", jac=True, options={"disp": True}
```
Comment (proteneer) [inline timemachine/md/minimizer.py:355]: it turns out that `L-BFGS-B` is actually quite unstable. Reverting this.

**Comment by proteneer** [comment]
I reduced the search radius from `1.0nm` to `0.5nm` which sped things up drastically since it reduced the # of local atoms to about 300.

**Comment by proteneer** [comment]
github went FUBAR - going to rebase and open a new PR.

**Pull request closed without merging**